platform/platform-impl/src/com/intellij/openapi/vfs/newvfs/RefreshSessionImpl.java

```diff
@@ -122,10 +127,15 @@ void scan() {
         ((LocalFileSystemImpl)fs).markSuspiciousFilesDirty(workQueue);
       }
 
-      long t = 0;
+      long t = System.currentTimeMillis();
+      String refreshSessionInfo = String.format("work queue size: %s , file types: %s", workQueue.size(),
+                                                workQueue.stream().collect(
+                                                  Collectors.groupingBy(file ->
+                                                                          file.getExtension() != null ?
+                                                                          file.getExtension() : (file.isDirectory() ? "folder" : "NA"),
```

Comment:
if not folder and no extension let's not guess since hard to pick good group for such case

Reply:
Better do `file -> file.isDirectory() ? "directory" : Objects.requireNonNullElse(file.getExtension(), "n/a")`.
The order of conditions matter, because `VirtualFile#getExtension` is a text-based method and may return misleading results if a directory happen to have a dot in a name; and the result isn't cached, so asking once once is more environment friendly.

Reply:
requireNonNullElse is JDK9+ so to avoid issues with null using toString() instead

Reply:
I have lost track again which modules are at which level :(

Anyway, `"null"` is okay for files (files w/o an extension aren't very common among refresh roots), but mixing directories into the same counter makes the log look more cryptic than it should.

Reply:
there is no good "type" for directory and with 11k of jars there is no way to avoid grouping and this is what makes TRACE kind of useless